### PR TITLE
Implement ring and little fingers within iCub::iKin::iCubFinger Class

### PR DESCRIPTION
**Pull request by xEnVrE**
This PR adds support for ring and little fingers within `iCub::iKin::iCubFinger` Class.

### CODE REVIEW

**Comment by traversaro** [comment]
Cool. Do you have a reference from where you extracted the DH from?

File: src/libraries/iKin/src/iKinFwd.cpp [hunk 1]
```diff
@@ -1947,6 +1993,24 @@ bool iCubFinger::alignJointsBounds(const deque<IControlLimits*> &lim)
         (*this)[2].setMin(CTRL_DEG2RAD*min);
         (*this)[2].setMax(CTRL_DEG2RAD*max/2.0);
     }
+    if (finger=="ring" || finger=="little")
```
Comment (pattacini) [inline src/libraries/iKin/src/iKinFwd.cpp]: This should be:
```c++
else if (...)
```

**Comment by claudiofantacci** [comment]
@traversaro, I got the DH parameters from Parmiggiani who extracted them from the iCub schematics.

We double checked the parameters with the existing one and they are the same (apart from some small variation that we are going to check soon), plus @xEnVrE added the missing fingers.

Please note that there are still some problem on the thumb. By a simple investigation it seems that we are missing an offset on the thumb oppose encoder.

If you need any data, just let us know and we can provide further details.

**Comment by traversaro** [comment]
If you could write down the Creo assembly filename (should be in https://svn.robotology.eu/repos/iCubHardware-pub/trunk/mechanics/public/icub1_icub2/ I guess) and the SVN revision used for generating the DH even as comment in this PR, I think it would be really helpful for the future.

**Comment by claudiofantacci** [comment]
@traversaro I am really not aware of neither the Creo assembly filename, neither the possible SVN revision used for generating the DH.

The work has been done long time ago by Parmiggiani and I and at that time I was unaware of such details. I'll ask Parmiggiani whether we have the change to recover such infomation and let you know.

**Pull request merged**